packages/language-server/src/lib/FallbackWatcher.ts

```diff
@@ -18,32 +22,48 @@ export class FallbackWatcher {
                     gitOrNodeModules.test(path) &&
                     // Handle Sapper's alias mapping
                     !path.includes('src/node_modules') &&
-                    !path.includes('src\\node_modules')
+                    !path.includes('src\\node_modules'),
+
+                // typescript would scan the project files on init.
+                // We only need to know what got updated.
+                ignoreInitial: true,
+                ignorePermissionErrors: true
             }
         );
 
         this.watcher
-            .on('add', (path) => this.callback(path, FileChangeType.Created))
-            .on('unlink', (path) => this.callback(path, FileChangeType.Deleted))
-            .on('change', (path) => this.callback(path, FileChangeType.Changed));
+            .on('add', (path) => this.onFSEvent(path, FileChangeType.Created))
+            .on('unlink', (path) => this.onFSEvent(path, FileChangeType.Deleted))
+            .on('change', (path) => this.onFSEvent(path, FileChangeType.Changed));
     }
 
-    private convert(path: string, type: FileChangeType): DidChangeWatchedFilesParams {
-        const event: FileEvent = {
+    private convert(path: string, type: FileChangeType): FileEvent {
+        return {
             type,
             uri: pathToUrl(path)
         };
-
-        return {
-            changes: [event]
-        };
     }
 
-    private callback(path: string, type: FileChangeType) {
-        const para = this.convert(path, type);
-        this.callbacks.forEach((callback) => callback(para));
+    private onFSEvent(path: string, type: FileChangeType) {
+        const fileEvent = this.convert(path, type);
+
+        this.undeliveredFileEvents.push(fileEvent);
+        this.scheduleTrigger();
     }
 
+    private readonly scheduleTrigger = debounceSameArg<void>(
```

Comment:
We have 'lodash' as a dependency, you can simplify this and use their `debounce` function for this case